### PR TITLE
Add uint64 support to v2

### LINKED ISSUE #49: uint64 loses data when unmarshalling
Currently, any uint64 which overflows int64 gets unmarshalled as a float64 instead.

At best, this loses precision (since the float64 only has 52 bits to store the fraction in).

At worst, a bug in the boundary condition prevents the unmarshalling from happening at all:
`if resolved < 1<<64-1`

This should be `if resolved <= 1<<64-1`...

Test case:

```
package main

import (
    "fmt"
    "math"
    "strconv"

    "gopkg.in/yaml.v2"
)

func main() {
    fmt.Println("\nTesting loss of precision\n=======================================")
    test(12345678901234567890)

    fmt.Println("\nTesting uint64 max\n=======================================")
    test(math.MaxUint64)
}

type Test struct {
    X uint64
}

func test(x uint64) {
    m := Test{x}

    fmt.Printf("Original (base-2):      %s\n", strconv.FormatUint(m.X, 2))
    fmt.Printf("Original (base-10):     %s\n\n", strconv.FormatUint(m.X, 10))

    y, _ := yaml.Marshal(m)
    fmt.Printf("Marshalled YAML:     %s\n", string(y[:]))

    var m_yaml Test
    yaml.Unmarshal(y, &m_yaml)
    fmt.Printf("Unmarshalled (base-2):  %s\n", strconv.FormatUint(m_yaml.X, 2))
    fmt.Printf("Unmarshalled (base-10): %s\n\n", strconv.FormatUint(m_yaml.X, 10))
}
```

### PR DESCRIPTION
**Pull request by liggitt**
Resolves #49 for v2

### CODE REVIEW

**Comment by liggitt** [comment]
@niemeyer removed uint cases, left uint64 handling. Also inlined testcases as literal values to make it easier to follow, and added 64-bit support to the binary int decoding.

File: decode_test.go [hunk 1]
```diff
@@ -267,6 +267,114 @@ var unmarshalTests = []struct {
 		map[string]uint64{},
 	},
 
+	// int
+	{
+		"int_max: 2147483647", // math.MaxInt32
```
Comment (niemeyer) [inline decode_test.go]: Instead of having comments on all these entries, it might be better and more accurate to just use the actual value in the assertion. For example:

```
"int_max: 2147483647",
map[string]int{"int_max": math.MaxInt32},
```

**Comment by niemeyer** [comment]
This looks good and I'm happy to merge it. Just see what you think of the suggestion above please.

File: decode_test.go (hunk 1 above)
Comment (liggitt) [inline decode_test.go]: I like it, will update

**Comment by liggitt** [comment]
@niemeyer ready for merge

**Comment by niemeyer** [comment]
Jordan, I got a bit ahead of myself here. I should have asked you to sign the contributor agreement first:

http://www.ubuntu.com/legal/contributors

It's a straightforward agreement, easy to sign online. Can you please have a look and see if it makes sense?  We need that for contributions.

**Comment by liggitt** [comment]
Let me check, and I'll let you know as soon as I get the go ahead

**Comment by niemeyer** [comment]
I'll need to back out the change if I don't get the signature this week. Sorry for the trouble, but this is required and I screwed up by not requesting beforehand.

**Comment by niemeyer** [comment]
Actually, to avoid creating a larger problem for people that update now, I have backed out the change right now. Please just ping me once you get the signature in place, and I'll happily reapply.

**Comment by liggitt** [comment]
No problem, I'll update as soon as I can

**Comment by niemeyer** [comment]
Thanks, and sorry for the trouble.

**Comment by niemeyer** [comment]
Any news there?

**Comment by liggitt** [comment]
Haven't forgotten... still waiting to hear back, sorry

**Comment by liggitt** [comment]
@niemeyer just got approval and signed the contributor agreement. Sorry for the delay.

**Comment by liggitt** [comment]
@niemeyer you want me to open another PR or do you want to re-open/merge this one?

**Comment by niemeyer** [comment]
Please leave it with me. I'll just reapply the change.

Thanks for handling the bureaucracy.

**Comment by liggitt** [comment]
@niemeyer thanks... I'd already spawned #53, in case that makes it easier
